### DISCUSSION TITLE
New organizer for new users?

### DISCUSSION THREAD

**Discussion by dkelsen** [pretix/pretix#4342] [Q&A] [answered]
Hi there,

I'm exploring the Pretix community edition and I'm trying to figure out the following:

When someone signs up in through `/register`, the new user is created without being part of an organizer. As a result, they're not able to do create any eents. Is there any built-in functionality that creates an organizer for the user, too? Or is the only way to create a new organizer as an admin (which they would not be and should not be in my use case)?

**Comment by raphaelm** [comment, marked as answer]
There is no functionality to create a new organizer in pretix community, no.

**Reply by MohamedEihab** [reply]
> There is no functionality to create a new organizer in pretix community, no.

@raphaelm Out of interest, is there a particular reason why that functionality is not implemented? Just wondering if it due to a technical implementation difficulty or another reason.

**Reply by raphaelm** [reply]
I don't think it makes much sense? Usually, self-hosted pretix is meant for "internal usage", e.g. company A installs pretix to manage their own events. Opening organizer self-registration now would be very unwanted since any internet user could open an account to host events on the server of company A. That is usually not wanted.

**Reply by MohamedEihab** [reply]
Yea I suppose the edge case would be an company wants users within it's own company to be able to setup and own events and therefore be organisers. Kind of like a school campus with different clubs and societies and you would want to separate out the events while still being part of the overall self-hosted pretix. 

Looking at the code, I don't think it would be too difficult to add this to the API. Unless I am mistaken, creating an organiser performs two actions - creating the organiser itself, and creating an admin team for that organiser with the user being apart of the team.

**Reply by raphaelm** [reply]
Adding it to the API is hard because most methods of API authentication are "within" an organizer. But it's possible, even as a plugin – after all, that is how pretix.eu works ;)

**Reply by raphaelm** [reply]
> Yea I suppose the edge case would be an company wants users within it's own company to be able to setup and own events and therefore be organisers.

Yeah but how do you control only internal people set up accounts? Usually easy enough to just have some central team create the organizer and invite the first admin for it.

**Reply by MohamedEihab** [reply]
> Yeah but how do you control only internal people set up accounts? Usually easy enough to just have some central team create the organizer and invite the first admin for it.

If we use the university example above, I think then the user is already vetted and verified and has a university email for example / account. This can be used to ensure that only those accounts are able to create a single organizer related to their account but this purely just a thought experiment.

> Adding it to the API is hard because most methods of API authentication are "within" an organizer. But it's possible, even as a plugin – after all, that is how pretix.eu works ;)

True it's a bit of an odd flow, maybe something along the lines of create an organizer somehow, then for that user generate a token from the admin team that they can use for the rest of the API operation.

But yea this clears up why creating an organizer is unavailable for the API thank you :)

**Reply by dkelsen** [reply]
How is it implemented on pretix.eu? I believe you can sign up with a new account and an organizer will be created for the account.

**Reply by raphaelm** [reply]
Yes. It's a custom plugin that adds views that create the required model instances.
